docs/source/distributed.fsdp.fully_shard.rst

```diff
@@ -0,0 +1,66 @@
+torch.distributed.fsdp.fully_shard
+==================================
+
+PyTorch FSDP2 (``fully_shard``)
+-------------------------------
+
+PyTorch FSDP2 provides a fully sharded data parallelism (FSDP) implementation
+targeting performant eager-mode while using per-parameter sharding for improved
+usability. Compared to PyTorch FSDP1 (``FullyShardedDataParallel``):
+
+- FSDP2 uses ``DTensor``-based dim-0 per-parameter sharding for a simpler
```

Comment:
nit: should we define what 'per-parameter sharding' actually is?  (mention how copy-in is used, etc)  - or redirect to another wiki where this is explained more in depth?